src/Rings/mpoly-ideals.jl

```diff
@@ -1103,13 +1107,56 @@ end
 function minimal_primes(
     I::MPolyIdeal{T}; 
     algorithm::Symbol=:GTZ, 
+    factor_generators::Bool=true,
+    simplify_ring::Bool=true,
     cache::Bool=true
   ) where {U<:Union{AbsSimpleNumFieldElem, <:Hecke.RelSimpleNumFieldElem}, T<:MPolyRingElem{U}}
   has_attribute(I, :minimal_primes) && return get_attribute(I, :minimal_primes)::Vector{typeof(I)}
 
   R = base_ring(I)
+  is_one(I) && return typeof(I)[]
+
+  # Try to eliminate variables first. This will often speed up computations significantly.
+  if simplify_ring
+    Q, pr = quo(R, I)
+    W, id, id_inv = simplify(Q)
+    @assert domain(id) === codomain(id_inv) === Q
+    @assert codomain(id) === domain(id_inv) === W
+    res_simp = minimal_primes(modulus(W); algorithm, factor_generators, simplify_ring=false)
+    result = [I + ideal(R, lift.(id_inv.(W.(gens(j))))) for j in res_simp]
+    for p in result
+    end
+    return result
+  end
+
+  # This will in many cases lead to an easy simplification of the problem
+  if factor_generators
+    J = typeof(I)[ideal(R, elem_type(R)[])]
+    for g in gens(I)
+      K = typeof(I)[]
+      is_zero(g) && continue
+      for (b, k) in factor(g)
+        for j in J
+          push!(K, j + ideal(R, b))
+        end
+      end
+      J = K
+    end
+    result = unique!(filter!(!is_one, vcat([minimal_primes(j; algorithm, factor_generators=false) for j in J]...)))
+    # The list might not consist of minimal primes only. We have to discard the embedded ones
+    final_list = typeof(I)[]
+    for p in result
+      any((q !== p && is_subset(q, p)) for q in result) && continue
+      push!(final_list, p)
+    end
+    for p in final_list
+      set_attribute!(p, :is_prime=>true)
+    end
+    return final_list
+  end
```

Comment:
@afkafkafk13 , @hannes14 : Could you let me knew what you think of these? I tried to run a script again from about 5 months ago and it broke down on some primary decompositions (which used to go though at some point). Now I'm trying to get back there with these tweaks. And I thought that they might be useful in general. 

@hannes14 : This was also used to prepare the example I sent you. Interestingly it is solved rather quickly with `algorithm=:charSets`, but never terminated for me with `:GTZ`.